streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1598,7 +1598,6 @@ private long countStreamThread(final Predicate<StreamThread> predicate) {
      * @return the set of {@link ThreadMetadata}.
      */
     public Set<ThreadMetadata> localThreadsMetadata() {
-        validateIsRunningOrRebalancing();
```

Comment:
If this function could bypass rebalancing state check, what about other functions such as `allMetadata`?